segmentation/utils/signal_extraction.py

```diff
@@ -1,36 +1,89 @@
+from skimage.measure import regionprops
 
 
-def simple_weighted_extraction(cell_coords, image_data):
+# TODO: work on weighted extraction and implement other discussed techniques of extraction
+def positive_pixels_extraction(cell_coords, image_data, threshold=0):
     """
-    Define a weighted extraction technique based on how close the coordinate is to the center
+    Extract channel counts by summing over the number of non-zero pixels in the cell,
+    improves on default_extraction by not distinguishing between pixel expression values
 
     Args:
-        cell_coords: tuples of values representing pixels within one cell
-        image_data: array containing channel counts
+        cell_coords (numpy): values representing pixels within one cell
+        image_data (xarray): array containing channel counts
 
     Returns:
-        np.array: sum of counts for each channel
+        channel_counts (numpy): sum of counts for each channel
     """
 
-    pass
+    # index indo image_data to get th channel values we're interested in
+    channel_values = image_data.values[tuple(cell_coords.T)]
+
+    # sum up based on a binary mask that is 1 if the expression value > threshold else 0
+    channel_counts = np.sum(channel_values > threshold, axis=0)
+
+    return channel_counts
+
+
+def center_weighting_extraction(cell_coords, image_data):
+    """
+    Extract channel counts by summing over weighted expression values based on distance from center,
+    improves upon default extraction by including a level of certainty/uncertainty
+
+    Args:
+        cell_coords (numpy): values representing pixels within one cell
+        image_data (xarray): array containing channel counts
+        segmentation_mask (numpy): array containing segmentation labels for each cell
+        cell_label (int): the cell number we wish to extract for each cell, important to distinguish
+            which centroid we want in regionprops because this function only extracts signal
+            for one cell, though we may choose to change this in the future...
+
+    Returns:
+        channel_counts (numpy): sum of counts for each channel
+    """
+
+    # we could use regionprops but it's probably better to use this method (ala A-Kag's)
+    # because it doesn't require you to know which cell_label to index into
+    centroid = np.sum(cell_coords, axis=0) / cell_coords.shape[0]
+    centroid = centroid.astype(np.int16)
+
+    # this will ensure we never get zero or negative weighting values in our weight matrix
+    # I'm currently rounding but it's sort of arbitrary, may decide to truncate (aka round down)
+    # in case of decimal centroid values returned by regionprops
+    center_weight = max(round(centroid[0]), round(centroid[1])) + 1
+
+    # TODO: this is still not a perfect way to do this for even image dimensions
+    # but I think Adam's suggested method might fix this
+    image_row_index, image_col_index = np.ogrid[:image_data.shape[0], :image_data.shape[1]]
```

Comment:
This is a bit inefficient. The image_data array is going to typically be an array of 1024x1024. Most cells are going to around 30x30 pixels. We will likely have ~500 cells per image.

I'm sure ogrid is pretty fast, but it seems like there could be a better way to do this? I'm guessing regionprops will help here. For example, you can get the coordinates of the bounding box that encloses the cell.

Reply:
regionprops_table could definitely help in reducing the partition of the image np.ogrid looks at. Adam also had a suggestion that I'll look into.

Reply:
adam's suggestion is pretty neat.  but for either one, passing in both `centroid=local_centroid` and `image_data=intensity_image * image` (* is element wise here) can replace centroid and channel_values respectively.  Furthermore, that completely eliminates the need for passing in cell-coords+indexing at the end, since:
```
...

channel_values = image_data
weight_values = np.expand_dims(weight_matrix, axis=0)

channel_counts = weight_values.dot(channel_values)
```
Same with my version, but w/o cell_coords, an extra step is needed to get local coords:
```
...

dist_vecs = np.indices(image_data.shape).resize(2,-1).T - centroid

distances = np.linalg.norm(dist_vecs, axis=1)

weights = 1 - (distances/(np.max(distances) + 1))

channel_counts = weights.dot(image_data)
```